### PR TITLE
Tweaks for DraggableWin directive

### PR DESCRIPTION
**Pull request by ntma**
Fixes:
* [x] Support for Firefox (e.path is Google Chrome specific);
* [x] Reset dragger offset for each mousedown event (prevents magical jumps);
* ~Constain floating windows to the main container;~ (future PR)

#252

### CODE REVIEW

**Comment by fschmenger** [comment]
@ntma Hi Nuno, thanks for kickstarting this! I'm currently a little limited in time but can give it a test run by next week hopefully. 

Keep in mind that Task 3) will be a breaking change due to the changes in positioning offsets. Probably this should be in a separate PR, so it can be incorporated with the next Wegue major version.

**Comment by ntma** [comment]
Hi @fschmenger , no problem!

> Keep in mind that Task 3) will be a breaking change due to the changes in positioning offsets. Probably this should be in a separate PR, so it can be incorporated with the next Wegue major version.

Indeed, I got  to agree with that. I will take your suggestion!

File: src/directives/DraggableWin.js [hunk 1]
```diff
@@ -108,11 +107,18 @@ export const DraggableWin = {
...
+      // get the element path
+      const path = e.path || (e.composedPath && e.composedPath());
+
       if (_data.draggableElementSelector &&
-          !checkIfSelectorInPath(_data.draggableElementSelector, e.path)) {
+          !checkIfSelectorInPath(_data.draggableElementSelector, path)) {
         return;
       }
 
+      // set the offset for each moousedown
```
Comment (JakobMiksch) [inline src/directives/DraggableWin.js:119]:
```suggestion
      // set the offset for each mousedown
```

**Review by JakobMiksch — COMMENTED** [review]
I do not really understand the change here. Could you explain what the behavior before and after this PR is?

File: src/directives/DraggableWin.js (hunk 1 above)
Comment (ntma) [reply src/directives/DraggableWin.js:119]: Hi @JakobMiksch,

To reproduce the issue, open two windows configured with `draggable: true` and drag one window after the other

**Before this PR**, the current window being dragged will always offset to the position of the previously dragged window.

**After this PR**, the offset is resetted on each drag start.

This issue is caused by the following object being shared across all components that use the `DraggableWin.js` directive.
https://github.com/wegue-oss/wegue/blob/7f53a55545015d37c5b4fc73c8f9dba334240b7d/src/directives/DraggableWin.js#L9-L26

Comment (JakobMiksch) [reply src/directives/DraggableWin.js:119]: Alright. I have actually never used this dragging functionality so far. 
I only could drag the InfoClick window so far. The other ones do not work to drag. 
Do you know why, which windows do you use for testing?

Comment (ntma) [reply src/directives/DraggableWin.js:119]: I tested with `wgu-maprecorder` and the `wgu-infoclick` in google chrome. Had to set `draggable: true` to both of them in the app-conf.json.  Are you able to reproduce?

Comment (JakobMiksch) [reply src/directives/DraggableWin.js:119]: Yes, I was able to reproduce. Good catch.

**Review by JakobMiksch — APPROVED** [review]
Looks fine from my side. I tested it and it does what it should.

**Comment by JakobMiksch** [comment]
@fschmenger maybe could have a look and merge if you approve.

File: src/directives/DraggableWin.js (hunk 1 above)
Comment (chrismayer) [reply src/directives/DraggableWin.js]: Other windows like the measure window do not get draggable even with the `draggable: true` in the `app-conf.json`. Reason is the positioning in the projects CSS file, where already a TODO is noted, that the positioning has to be harmonized, see https://github.com/wegue-oss/wegue/blob/master/src/assets/css/wegue.css#L112

**Comment by chrismayer** [comment]
Thanks @ntma for this. LGTM :+1: As @JakobMiksch suggested, let's see what @fschmenger says and if he approves feel free to merge afterwards.

**Comment by fschmenger** [comment]
OK, this looks good to merge. I can finally see the `draggable` option working in Firefox and it no longer get's confused when dragging multiple cards. 

I suggest we leave #252 open, until the other topics can be addressed for v2.

**Pull request merged**